### ISSUE TITLE
High cpu usage

### ISSUE THREAD

**Issue by sulavmhrzn** [microsoft/pylance-release#3554]
Type: <b>Performance Issue</b>

CPU usage fluctuates erratically and frequently remains high.

/usr/share/code/code --ms-enable-electron-run-as-node /home/sulav/.vscode/extensions/ms-python.vscode-pylance-2022.10.40/dist/server.bundle.js --cancellationReceive=file:3eda896e6ccf04f8a876b2515eed7b7462dcb4e7c9 --node-ipc --clientProcessId=121885
25
250
122016

Extension version: 2022.10.40
VS Code version: Code 1.72.2 (d045a5eda657f4d7b676dedbfa7aab8207f8a075, 2022-10-12T22:16:26.920Z)
OS version: Linux x64 5.19.16-301.fc37.x86_64
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz (4 x 3100)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 1, 1|
|Memory (System)|7.63GB (2.84GB free)|
|Process Argv|--unity-launch|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|wayland|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   164	121674	code main
    0	    31	121737	   zygote
    0	   117	121770	     gpu-process
    0	    31	121738	   zygote
    0	     0	121740	     zygote
    0	    86	154889	       issue-reporter
    0	    55	121777	   utility-network-service
    0	   281	121796	   window (auth.py - rent-my-book - Visual Studio Code)
    0	   141	121884	   shared-process
    0	    78	121911	     ptyHost
    0	     0	122011	       /usr/bin/zsh -i
    0	    55	122178	         /usr/bin/python3 -sP /usr/bin/poetry shell
    0	     0	122192	           /usr/bin/zsh -i
    0	    39	152171	             python3 main.py
    0	     8	152172	               /home/sulav/.cache/pypoetry/virtualenvs/rent-my-book-UwXx-c0e-py3.11/bin/python3 -c from multiprocessing.resource_tracker import main;main(4)
    0	    39	154853	               /home/sulav/.cache/pypoetry/virtualenvs/rent-my-book-UwXx-c0e-py3.11/bin/python3 -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=5, pipe_handle=7) --multiprocessing-fork
    0	    70	121977	     fileWatcher
    0	     0	154905	     /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	   141	121885	   extensionHost
   25	   250	122016	     /usr/share/code/code --ms-enable-electron-run-as-node /home/sulav/.vscode/extensions/ms-python.vscode-pylance-2022.10.40/dist/server.bundle.js --cancellationReceive=file:3eda896e6ccf04f8a876b2515eed7b7462dcb4e7c9 --node-ipc --clientProcessId=121885
    0	    62	122022	     /usr/share/code/code --ms-enable-electron-run-as-node /usr/share/code/resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=121885
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (auth.py - rent-my-book - Visual Studio Code)
|    Folder (rent-my-book): 21 files
|      File types: pyc(7) py(7) sqlite3(1) lock(1) toml(1)
|      Conf files:;
```

</details>

**Comment by Alekky09** [comment]
Exact same thing happening for me in macOS Big Sur 11.3.1

**Comment by mrucker** [comment]
This has recently become a problem for me too

**Comment by rchiodo** [comment]
Thanks for the issue. This should be fixed in the prerelease version (2022.10.41)
